### PR TITLE
Updating tensorflow jvm

### PR DESCRIPTION
**Pull request by lastmansleeping**
This PR isolates the changes to upgrade the `ml4ir-inference` tensorflow jars to the latest serving libraries available. 

**Note** -> I've deliberately filed this PR against `master` to show that this new version can serve models trained on the old tensorflow==2.0.4 python stack.

### CODE REVIEW

**Comment by lastmansleeping** [comment]
@armgiraud Do you know how I can upgrade the jar version from `0.0.4` to `0.0.5`?

File: jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala [hunk 1]
```diff
@@ -1,73 +1,63 @@
...
-  *                ├── assets
-  *                ├── saved_model.pb
-  *                └── variables
-  *                    ├── variables.data-00000-of-00001
-  *                    ├── variables.data-00000-of-00002
-  *                    ├── variables.data-00001-of-00002
-  *                    └── variables.index
-  * @param config instantiated struct with the values from the feature_config.yaml relevant for inference
-  */
+ * @see <a href="https://www.tensorflow.org/api_docs/java/reference/org/tensorflow/SavedModelBundle">SavedModelBundle</a>
```
Comment (ArmandGiraud) [inline jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala:10]: Why do we need those changes?

File: jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala [hunk 2]
```diff
@@ -1,73 +1,63 @@
...
-          val predictions: Array[Array[Array[Float]]] =
-            Array
-              .ofDim[Float](resultTensor.shape()(0).toInt, resultTensor.shape()(1).toInt, resultTensor.shape()(2).toInt)
-          resultTensor.copyTo(predictions)
-          predictions(0)(0)
-        case _ =>
-          throw new IllegalArgumentException("unsupported result shape: " + resultTensor.shape())
+        .asInstanceOf[TFloat32]
+      resultTensor.shape().numDimensions() match {
+        case 1 => StdArrays.array1dCopyOf(resultTensor)
```
Comment (ArmandGiraud) [inline jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala:54]: This is much simpler!

**Review by ArmandGiraud — APPROVED** [review]
LGTM

File: jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala (hunk 1 above)
Comment (lastmansleeping) [reply jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala:10]: I think this was just an indentation change from the IDE

**Pull request merged**